From 0243ab89cd941f15344ef03b63c96f3f0e1ffa45 Mon Sep 17 00:00:00 2001
From: Brandon Philips <brandon@ifup.co>
Date: Tue, 1 Apr 2014 11:51:13 -0700
Subject: [PATCH] fix(google-daemon): use * for the passwd not !

useradd defaults to using ! as the "locked" password marker.
Unfortunatly, openssh interprets this to mean that it shouldn't let the
user in via ssh if PAM is missing. Work around this by using the *
marker which also means locked but is allowed by openssh.
---
 google-daemon/usr/share/google/google_daemon/utils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/google-daemon/usr/share/google/google_daemon/utils.py b/google-daemon/usr/share/google/google_daemon/utils.py
index 0f17f34..371034d 100755
--- a/google-daemon/usr/share/google/google_daemon/utils.py
+++ b/google-daemon/usr/share/google/google_daemon/utils.py
@@ -69,7 +69,7 @@ class System(object):
   def UserAdd(self, user, groups):
     logging.info('Creating account %s', user)
     result = self.RunCommand([
-        '/usr/sbin/useradd', user, '-m', '-s', '/bin/bash', '-G',
+        '/usr/sbin/useradd', user, '-m', '-s', '/bin/bash', '-p', '*', '-G',
         ','.join(groups)])
     if self.RunCommandFailed(result, 'Could not create user %s', user):
       return False
-- 
1.8.5.2 (Apple Git-48)

